### PR TITLE
tests/sys/progress_bar: move configuration to app.config

### PR DESCRIPTION
**Pull request by mguetschow**
### Contribution description

Providing the configuration via Makefile variables if Kconfig is supported by the module may involve some non-determinism in the environment variable that is finally exposed to the python tests. This commit moves the configuration from Makefile to app.config.

### Testing procedure

Confirm that the configuration from `app.config` is applied locally with `make -C tests/sys/progress_bar all test`

Merge and wait for the next nightly with fingers crossed.

### Issues/PRs references

Follow-up of #20763 and will need to follow-up to remove the debug prints in the python script.

### CODE REVIEW

File: tests/sys/progress_bar/tests/01-run.py [hunk 1]
```diff
@@ -18,14 +18,14 @@
 
 
 def testfunc(child):
-    for i in range(0, 100, 10):
+    for i in range(0, 101, 10):
```
Comment (miri64) [inline tests/sys/progress_bar/tests/01-run.py:21]: Can you factor this out into its own commit, please? This seems to be unrelated to the Kconfig fix.

Comment (mguetschow) [reply tests/sys/progress_bar/tests/01-run.py:21]: Sure, done

**Review by miri64 — APPROVED** [review]
Let's see if this helps.
